syncthing-git/PKGBUILD

```diff
@@ -5,63 +5,66 @@
 # Previous Maintainer : Sauyon Lee <sauyonl@sauyon.com>
 # Contributor : Martin Wimpress <code@flexion.org>
 
-pkgname=syncthing-git
-pkgver=v1.5.0.rc.2.r18.g037934ec7
+_pkgname=syncthing
+pkgname=$_pkgname-git
+pkgver=v1.6.0.rc.4.r21.g668979605
 pkgrel=1
 pkgdesc="Open Source Continuous Replication / Cluster Synchronization Thing"
-url="http://syncthing.net/"
+url="https://syncthing.net/"
 license=('MPLv2')
 makedepends=('git' 'go' 'inetutils')
-conflicts=('syncthing')
-provides=('syncthing')
+conflicts=("$_pkgname")
+provides=("$_pkgname")
 arch=('i686' 'x86_64')
-source=("$pkgname::git+https://github.com/syncthing/syncthing.git")
+source=("git+https://github.com/syncthing/syncthing.git")
 sha256sums=('SKIP')
-install=${pkgname}.install
-_name=syncthing
+install=${_pkgname}.install
 
 prepare() {
-	cd "${srcdir}"
+	cd "$srcdir/$_pkgname"
+	mkdir -p "${srcdir}/../go"
+	export GOPATH="${srcdir}/../go"
+	export PATH="$PATH:$GOPATH/bin"
+	# fix for broken version
+	chmod u+w -R "$GOPATH"
+	go get -d -v ./...
 }
 
 pkgver() {
-	cd "${pkgname}"
+	cd "$srcdir/$_pkgname"
 	git describe --long --tags | sed 's/\([^-]*-g\)/r\1/;s/-/./g'
 }
 
 build() {
-	export GOPATH="${srcdir}/${pkgname}-${pkgver}"
-	cd "${srcdir}"
-	mkdir -p "src/github.com/syncthing"
-	mv "${pkgname}" "src/github.com/syncthing/${_name}"
-	cd "src/github.com/syncthing/${_name}"
-	if [ ${CARCH}" == "i686" ] ; then
-			go run build.go -no-upgrade -goarch 386 build
-	if [ ${CARCH}" == "x86_64" ] ; then
-			go run build.go -no-upgrade -goarch amd64 build
-	else
-			go run build.go -no-upgrade build
-	fi
+	cd "$srcdir/$_pkgname"
+	export GOPATH="${srcdir}/../go"
+	export PATH="$PATH:$GOPATH/bin"
+	export CGO_LDFLAGS="${LDFLAGS}"
+	export CGO_CFLAGS="${CFLAGS}"
+	export CGO_CPPFLAGS="${CPPFLAGS}"
+	export CGO_CXXFLAGS="${CXXFLAGS}"
+	export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
```

Comment:
Do you really need `-mod=readonly`? I think it's superficial since you already resolved the dependencies.

Reply:
I don't use go and therefore have no real knowledge about such things. I would prefer keeping it for now because as far as I understand it, it shouldn't do any harm.